Improve basics aa3 to implement 1

pinterid opened the pull request:
**Checklist:**

 <!--- Make sure your PR is documented and tested before submission. Put an `x` in all the boxes that apply: -->
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?


**Type of Changes:**

 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Documentation (non-breaking change which adds documentation)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)


**What is the current behavior? (link to any open issues here)**
- Currently the active tab of a software profile is not remembered. #72 
- Currently the sign in button in the navigation bar redirects to the start page, but not directly to the sing in form. #73
- Currently it is not possible to upload custom avatar images. #76 
- Currently user settings can be changed via a modal. #77 
- Currently when recaching data the avatar image resets. #78 
- Currently the application structure does not fit our needs because several issues. #80
- Currently there is no notification when sining in fails. #81 
- Currently the pinned repositories are kind of useless because they do not show any useful information. #82 
- Currently there are some issues in styling when displaying projects. #83 
- Currently there is no language chart for projects. #84 
- Currently there is no default image for projects without an avatar image/owner. #85 
- Currently there is no language chart for a user. #86
- Currently a user cannot modify the position of charts on his own overview page. #88 
- Currently the design seems a bit outdated and there are several minor issues. #90 



**What is the new behavior (if this is a feature change)?**
- Now the active tab of a software profile is remembered. #72 
- Now the navbar login button redirects to the login form. #73 
- Now it is possible to upload custom avatar images. #76 
- Now user settings can be changed via a separate settings page. #77 
- Now when uploading a avatar image it is permanent. #78 
- Now redux is implemented which resolves issues like active tabs when state changes,.. #80 
- Now when singing in fails there is a notification in the login form. #81 
- Now the pinned repositories are temporary disabled. #82 
- Now the styling of projects is refined and also works fine with projects without an owner. #83 
- Now there is a language chart for each project. #84 
- Now there is a default image for projects without a an avatar/owner. #85
- Now there is a language chart on each user profile. #86
- Now a user can modify the position of charts on his own overview page. #88
- Now the design is more rounded and improved in general. Project styling, navbar styling,.. has been updated. #90

**Other information**:
- Ref: 🐍

pinterid committed:
Update globalState

The active object was added to globalState.

pinterid committed:
Implement active tab

The active software tab is now stored in the new globalState.

pinterid committed:
Implement active action card

The active actionCard is now stored in the new globalState.

pinterid committed:
Update navbar login

The navbar login button now sets the active actionCard.

Aichnerc reviewed (approved):
LGTM

schettn committed:
Refine signin button redirect and remove software tab workaround

With the introduced implementation it is possible to use the "Sign In"
button to redirect to the correct login action card. The workaround which allows reloading the last tab before
caching remounts every component is removed now.
The functionality which is missing now will be added soon.

schettn committed:
Upgrade to redux

New redux structure was roughly implemented. Refinements take place in the following commits. But .. redux is fun :)

schettn committed:
Enhance Redux

Some bugs are fixed and the actions were adjusted.

schettn committed:
Improve user fetching

Incorrect user fetching has been fixed.

pinterid committed:
Update package.json

React-Avatar-Editor was added for editing images on the site.

pinterid committed:
Add profile picture modal

The modal for editing images was added.

pinterid committed:
Update modal export/imports

The modal imports and exports were updated.

pinterid committed:
Add settings page

A page for editing user settings was added.

pinterid committed:
Implement settings page

The path for the settings page was set.

schettn committed:
Upgrade registration component with redux

The registration works now with the new structure.

pinterid committed:
Update navbar settings

The navbar settings now work with the settings page.

pinterid committed:
Update settings action

The settings action is now capable of updating profile pictures.

schettn committed:
Login after regisration

A context error where the login function was not defined due to asynchronously
is fixed now. After registering a user will be logged in.

schettn committed:
Improve settings page accesability

Now it is possible to access the settings page directly by entering the url. Access is only granted to the own settings page.

schettn committed:
Improve user menu

Now when clicking on `My Profile` only the corresponding components are rendered and not longer everyone like before.

schettn committed:
Remove legacy code

Due to the switch to Redux, a lot of code is obsolete and is now being removed.

schettn committed:
Merge branch 'improve-settings-aa1' into add-redux-aa1

schettn committed:
Improve Redux DevTools integration

Now  Redux DevTools only applies in development mode.

schettn committed:
Fix syntax error of previous merge

The resolving of the merge conflicts went wrong and produced a syntax error which
is fixed now.

schettn committed:
Conditional Rendering of Talk Interaction Buttons

Uploading and deleting of talks is now only possible if a user is signed in
and on his profile.

pinterid committed:
Remove init blank

Blank initialization was removed because it is no longer needed.

pinterid committed:
Add input value check

The check was added to prevent the uncontrolled component warning.

schettn committed:
Merge remote-tracking branch 'origin/improve-settings-aa2' into add-redux-aa1

Aichnerc committed:
Add Language chart

The chart for displaying languages has been added.

Aichnerc committed:
Improve projects

The project component has been improved and now actually looks nice.

Aichnerc committed:
Improve profile info

The language chart and top languages have been implemented.

Aichnerc committed:
Disable pinned

The pinned posts have been temporarily disabled.

schettn committed:
Merge branch 'improve-design-ba1' into add-redux-aa1

pinterid committed:
Update autch actions

The auth actions errors were updated.

pinterid committed:
Update login form errors

The login form errors were updated.

schettn committed:
Merge branch 'add-notifications-aa1' into add-redux-aa1

schettn commented:
Redux?

schettn committed:
Add Default Project Avatar

When there is no image available for a project, a default image is set.

schettn committed:
Improve Default Major Language Name

The default text has been adjusted cause of the suck-level of the previous one.

schettn committed:
Add Todo To LanguageChart Component

A todo has been added. It includes renaming of component and descriptions
due to a copy paste issue...

schettn committed:
Merge branch 'improve-design-ba2' into add-redux-aa1

schettn committed:
Update SignIn Error Message

The message have been refined cause the old one was a bit shitty.

schettn committed:
Add Redirect to Homepage in Settings page for unlogged users

Now when a anonymous user visits the settings page he is redirected to
the homepage with active login form.

pinterid committed:
Fix Codacy

Issues marked by Codacy were resolved.

pinterid committed:
Fix Codacy

Issues marked by Codacy were resolved.

pinterid committed:
Fix Codacy

Codacy errors were fixed.

schettn committed:
Move Changeable Content to User Section

Previously some data was stored in the profile section. Therefore when adjusting
data in settings it got overwritten during a refetch.
By moving this data to user section it does not get overwritten anymore.

schettn committed:
Add Optional Chaining to Login Result

Now when the result is undefined nothing happens cause there is no error.

schettn committed:
Fix Redirect to Homepage in Settings Page

There was an error where it always redirected if the settings url was entered
directly. This issue is fixed now,

schettn committed:
Remove Workaround

There was a workaround to update user settings. This is removed now because it
resets the user data every read cache iteration.

schettn committed:
Merge pull request #87 from snek-at/add-redux-aa1

Add redux aa1 to improve basics aa3

schettn committed:
Merge branch 'implement-1' into improve-basics-aa3

schettn committed:
Update LanguageChart Namings

Issues which occurred due to copy past errors are fixed now. This includes new
naming and descriptions.

schettn committed:
Improve Import and Export Regions

The cleanliness of import and export region was not sufficient
and was therefore improved.

schettn committed:
Remove Default Class Comment from Export Regions

The comment `Default Class` is completely useless and therefore removed.

schettn committed:
Add Comments to Export Regions

All components that needed proper export documentation have been updated.
Contains a description of withRouter and connect behaviour

schettn committed:
Logs Cleanup

All unnecessary console logs are removed now.

pinterid commented on src/Routes.js:
```diff
@@ -11,58 +11,27 @@ import {
-            />
-          )}
-        />
+        <Route exact path="/" component={(props) => <HomePage {...props} />} />
```
Sometimes {..props} is used and sometimes not, what is the difference, do some pages don not require props?

schettn replied:
This is because it should be a kind of template. If props should be handed over at some point, it works. However, it is currently not needed in any case. It is true that it is illogical why it is sometimes missing. Therefore it should be added to every route component.

pinterid commented on src/components/organisms/SoftwareTabs/index.jsx:
```diff
@@ -5,17 +5,32 @@ import React from "react";
 
 //> Components
 import { ProjectTab, OverviewTab, TalksTab } from "../tabs";
+
```
```suggestion
```

pinterid commented on src/components/pages/TalkPage/index.jsx:
```diff
@@ -377,9 +374,20 @@ class TalkPage extends React.Component {
 }
 //#endregion
 
+const mapStateToProps = (state) => ({
+  selectedTalk: state.user.selectedTalk,
+});
+
+const mapDispatchToProps = (dispatch) => {
+  return { getTalk: (uid, username) => dispatch(getTalkAction(uid, username)) };
+};
+
```
```suggestion

//#region > Redux Mapping
const mapStateToProps = (state) => ({
  selectedTalk: state.user.selectedTalk,
});

const mapDispatchToProps = (dispatch) => {
  return { getTalk: (uid, username) => dispatch(getTalkAction(uid, username)) };
};
//#endregion

```

pinterid commented on src/store/actions/authActions.js:
```diff
@@ -0,0 +1,94 @@
+//#region > Authentication Actions
+/**
+ * Handle login
```
```suggestion
 * Handle login.
```

pinterid commented on src/store/actions/authActions.js:
```diff
@@ -0,0 +1,94 @@
+  return async (dispatch, getState, { getIntel }) => {
+    try {
+      const intel = getIntel();
+      const session = intel.snekclient.session;
```
```suggestion
      const session = intel.snekclient.session;
      
```

pinterid commented on src/store/actions/userActions.js:
```diff
@@ -0,0 +1,569 @@
+    try {
+      const intel = getIntel();
+      const state = getState();
+
```
```suggestion
```

schettn replied:
That's okay logical separation and so on..

pinterid reviewed (commented):
A few changes have to be made.

schettn committed:
Implement Loading Bar on Cache Update

Now there is a loading bar which gets displayed when updatingCache action is
running. In future, there will be a separate loading bar for this purpose.

schettn committed:
Improve Code Quality

Suggestions of @pinterid have been applied.

Aichnerc commented on src/components/atoms/Project/index.jsx:
```diff
@@ -7,6 +7,9 @@ import PropTypes from "prop-types";
 //> MDB
 // "Material Design for Bootstrap" is a great UI design framework
 import { MDBCol, MDBIcon } from "mdbreact";
+
```
Are new lines in imports a thing now?

schettn replied:
There is only one newline which separates extern and intern imports

Aichnerc commented on src/components/atoms/ScrollToTop/index.jsx:
```diff
@@ -29,7 +29,8 @@ class ScrollToTop extends React.Component {
 
 //#region > Exports
-//> Default Class
+// Got access to the history object’s properties and the closest
+// <Route>'s match.
```
This is not a multiline comment. See [tonic ](https://github.com/snek-at/tonic/blob/master/STYLE_GUIDE.md) on how to format multiline comments.

Aichnerc commented on src/components/atoms/SearchBar/index.jsx:
```diff
@@ -108,13 +106,26 @@ class SearchBar extends React.Component {
+// Provides its connected component with the pieces of the data it needs from
+// the store, and the functions it can use to dispatch actions to the store.
+// Got access to the history object’s properties and the closest
+// <Route>'s match.
```
This is not a multiline comment. Check [tonic](https://github.com/snek-at/tonic/blob/master/STYLE_GUIDE.md) on how to format a multiline comment.

Aichnerc commented on src/components/molecules/Footer/index.jsx:
```diff
@@ -254,7 +254,8 @@ class Footer extends React.PureComponent {
 //#region > Exports
-//> Default Class
+// Got access to the history object’s properties and the closest
+// <Route>'s match.
```
This is not a multiline comment. Please check [tonic](https://github.com/snek-at/tonic/blob/master/STYLE_GUIDE.md) on how to format multiline comments.

Aichnerc replied:
Ähm? Not the right formatting.

Aichnerc commented on src/components/molecules/Navbar/index.jsx:
```diff
@@ -59,136 +60,142 @@ class Navbar extends React.Component {
+// Provides its connected component with the pieces of the data it needs from
+// the store, and the functions it can use to dispatch actions to the store.
+// Got access to the history object’s properties and the closest
+// <Route>'s match.
```
This is not a multiline comment. Please check [tonic](https://github.com/snek-at/tonic/blob/master/STYLE_GUIDE.md) on how to format multiline comments.

Aichnerc commented on src/components/molecules/forms/LoginForm/index.jsx:
```diff
@@ -194,14 +201,24 @@ class LoginForm extends React.Component {
+// Provides its connected component with the pieces of the data it needs from
+// the store, and the functions it can use to dispatch actions to the store.
+// Got access to the history object’s properties and the closest
+// <Route>'s match.
```
This is not a multiline comment. Please check [tonic](https://github.com/snek-at/tonic/blob/master/STYLE_GUIDE.md) on how to format multiline comments.

Aichnerc commented on src/components/molecules/forms/RegisterForm/index.jsx:
```diff
@@ -825,15 +838,29 @@ class RegisterForm extends React.Component {
-//> Default Class
-export default RegisterForm;
+// Provides its connected component with the pieces of the data it needs from
+// the store, and the functions it can use to dispatch actions to the store.
```
This is not a multiline comment. Please check [tonic](https://github.com/snek-at/tonic/blob/master/STYLE_GUIDE.md) on how to format multiline comments.

kleberbaum commented on src/App.js:
```diff
@@ -1,16 +1,14 @@
-//> MDB
-// "Material Design for Bootstrap" is a great UI design framework
-import { MDBProgress } from "mdbreact";
+//> Redux
```
Add a Redux comment.

kleberbaum commented on src/App.js:
```diff
@@ -1,16 +1,14 @@
+//> Redux
+import { useDispatch } from "react-redux";
 
+//> Actions
```
Add a Actions comment.

kleberbaum commented on src/components/atoms/Project/index.jsx:
```diff
@@ -59,7 +65,6 @@ Project.propTypes = {
 //#endregion
 
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/atoms/ScrollToTop/index.jsx:
```diff
@@ -29,7 +29,8 @@ class ScrollToTop extends React.Component {
 //#endregion
 
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/atoms/SearchBar/index.jsx:
```diff
@@ -17,6 +15,11 @@ import {
 //> Fuzzysort
 // Fast SublimeText-like fuzzy search for JavaScript
 import * as fuzzysort from "fuzzysort";
+//> Redux
```
Add a Redux comment.

kleberbaum commented on src/components/atoms/SearchBar/index.jsx:
```diff
@@ -83,11 +80,12 @@ class SearchBar extends React.Component {
               fuzzysort
                 .go(this.state.filter, this.state.usernames)
                 .map((element, i) => {
+                  console.log("FUZZY ELEMNET", element);
```
Please remove or tag as DEBUG.

```suggestion
                  console.log("FUZZY ELEMNET", element);
```

kleberbaum commented on src/components/atoms/SearchBar/index.jsx:
```diff
@@ -108,13 +106,26 @@ class SearchBar extends React.Component {
+};
 //#endregion
 
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/atoms/charts/Calendar2D/index.jsx:
```diff
@@ -234,7 +234,6 @@ class Calender2D extends React.Component {
 //#endregion
 
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/atoms/charts/Calendar3D/index.jsx:
```diff
@@ -364,7 +364,6 @@ class Calendar3D extends React.Component {
 //#endregion
 
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/atoms/charts/ContribRadar/index.jsx:
```diff
@@ -180,7 +180,6 @@ class ContribRadar extends React.Component {
 //#endregion
 
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/atoms/charts/LanguageBar/index.jsx:
```diff
@@ -0,0 +1,66 @@
+}
+//#endregion
+
+//#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/atoms/charts/LatestActivity/index.jsx:
```diff
@@ -175,7 +175,6 @@ class LatestActivity extends React.Component {
 //#endregion
 
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/molecules/Footer/index.jsx:
```diff
@@ -254,7 +254,8 @@ class Footer extends React.PureComponent {
 //#endregion
 
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/molecules/Navbar/index.jsx:
```diff
@@ -17,21 +17,22 @@ import {
   MDBSmoothScroll,
   MDBBtn,
 } from "mdbreact";
+//> Redux
```
Add a Redux comment.

kleberbaum commented on src/components/molecules/Navbar/index.jsx:
```diff
@@ -59,136 +60,142 @@ class Navbar extends React.Component {
+};
+//#endregion
+
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/molecules/UserActionCard/index.jsx:
```diff
@@ -112,15 +108,7 @@ class UserActionCard extends React.Component {
-};
-//#endregion
-
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/molecules/forms/LoginForm/index.jsx:
```diff
@@ -4,12 +4,18 @@
+//> Additional
+// SHA hashing algorithm
+import { sha256 } from "js-sha256";
+//> Redux
```
Add a Redux comment.

kleberbaum commented on src/components/molecules/forms/LoginForm/index.jsx:
```diff
@@ -194,14 +201,24 @@ class LoginForm extends React.Component {
+};
+//#endregion
+
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/molecules/forms/RegisterForm/index.jsx:
```diff
@@ -29,9 +29,15 @@ import {
-
 //> OAuth
 import GitHubOAuth from "reactjs-oauth";
+//> Redux
```
Add a Redux comment.

kleberbaum commented on src/components/molecules/forms/RegisterForm/index.jsx:
```diff
@@ -825,15 +838,29 @@ class RegisterForm extends React.Component {
+};
+//#endregion
+
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/molecules/modals/ProfilePictureModal/index.jsx:
```diff
@@ -0,0 +1,148 @@
+}
+//#endregion
+
+//#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/organisms/ProfileInfo/index.jsx:
```diff
@@ -15,18 +15,23 @@ import {
   MDBIcon,
   MDBTooltip,
 } from "mdbreact";
+//> Redux
```
Add a Redux comment.

kleberbaum commented on src/components/organisms/ProfileInfo/index.jsx:
```diff
@@ -297,27 +287,79 @@ class ProfileInfo extends React.Component {
+};
+//#endregion
+
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/organisms/SoftwareTabs/index.jsx:
```diff
@@ -5,17 +5,32 @@ import React from "react";
 //> MDB
 // "Material Design for Bootstrap" is a great UI design framework
 import { MDBBadge } from "mdbreact";
+//> Redux
```
Add a Redux comment.

kleberbaum commented on src/components/organisms/SoftwareTabs/index.jsx:
```diff
@@ -5,17 +5,32 @@ import React from "react";
+  render() {
+    const { fetchedUser } = this.props;
+    const { activeTab } = this.state;
+
```
```suggestion
```

kleberbaum commented on src/components/organisms/SoftwareTabs/index.jsx:
```diff
@@ -111,9 +116,20 @@ class SoftwareTabs extends React.Component {
+};
+//#endregion
+
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/organisms/profiles/SoftwareEngineer/index.jsx:
```diff
@@ -41,7 +33,6 @@ class SoftwareEngineer extends React.Component {
 //#endregion
 
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/organisms/tabs/OverviewTab/index.jsx:
```diff
@@ -201,7 +201,6 @@ OverviewTab.propTypes = {
 //#endregion
 
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/organisms/tabs/ProjectTab/index.jsx:
```diff
@@ -42,7 +42,6 @@ ProjectTab.propTypes = {
 //#endregion
 
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/organisms/tabs/TalksTab/index.jsx:
```diff
@@ -16,11 +16,15 @@ import {
   MDBCardBody,
   MDBCardFooter,
 } from "mdbreact";
+//> Redux
```
Add a Redux commit.

kleberbaum commented on src/components/organisms/tabs/TalksTab/index.jsx:
```diff
@@ -74,14 +77,13 @@ class Talks extends React.Component {
       });
     }
-
     return (
```
```suggestion

    return (
```

kleberbaum commented on src/components/organisms/tabs/TalksTab/index.jsx:
```diff
@@ -118,14 +120,15 @@ class Talks extends React.Component {
                     </MDBCardHeader>
-                    <a
-                      href={
+
```
```suggestion
```

kleberbaum commented on src/components/organisms/tabs/TalksTab/index.jsx:
```diff
@@ -195,17 +198,34 @@ class Talks extends React.Component {
+};
+//#endregion
+
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/pages/CompanyPage/index.jsx:
```diff
@@ -668,7 +666,6 @@ class CompanyPage extends React.Component {
 //#endregion
 
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/pages/HomePage/index.jsx:
```diff
@@ -18,6 +18,8 @@ import {
   MDBBtn,
   MDBIcon,
 } from "mdbreact";
+//> Redux
```
Add a Redux comment.

kleberbaum commented on src/components/pages/HomePage/index.jsx:
```diff
@@ -301,9 +301,20 @@ class HomePage extends React.Component {
+};
+//#endregion
+
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/pages/TalkPage/index.jsx:
```diff
@@ -377,9 +374,20 @@ class TalkPage extends React.Component {
+  return { getTalk: (uid, username) => dispatch(getTalkAction(uid, username)) };
+};
+
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/components/pages/ProfilePage/index.jsx:
```diff
@@ -84,9 +126,30 @@ class ProfilePage extends React.Component {
+};
+//#endregion
+
 //#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

Aichnerc reviewed (changes requested):
There are a few things I've seen. Please check every file for comment formatting.

kleberbaum commented on src/store/reducers/authReducers.js:
```diff
@@ -0,0 +1,60 @@
+};
+//#endregion
+
+//#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/store/reducers/generalReducers.js:
```diff
@@ -0,0 +1,83 @@
+};
+//#endregion
+
+//#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/store/reducers/index.js:
```diff
@@ -0,0 +1,26 @@
+//#region > Imports
+//> Redux
```
Add a Redux comment.

kleberbaum commented on src/store/reducers/index.js:
```diff
@@ -0,0 +1,26 @@
+});
+//#endregion
+
+//#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum commented on src/store/reducers/userReducers.js:
```diff
@@ -0,0 +1,143 @@
+};
+//#endregion
+
+//#region > Exports
```
```suggestion
//#region > Exports
//> Default Class
```

kleberbaum reviewed (changes requested):
Should be ok to merge after a few changes.

schettn committed:
Add Movable Boundary

A boundary has been added which allows components to move within
a specified axis.

schettn committed:
Add Origin Check

A check which determines if the fetched and logged user matches.

schettn committed:
Integrate Movable in Overview Tab

A editor mode has been added which allows movement for all relevant
components.

schettn committed:
Add Export for Error Boundary

The missing export for error boundary has been added.

schettn committed:
Improve Styling for Overview Tab

The styling of sortable list and interchange charts has been improved.

Aichnerc committed:
Add scss to prettier

The SCSS file extension has been added to auto format for prettier.

Aichnerc committed:
Improve company page

The chart points have been removed.

Aichnerc committed:
Add color variables

Some more color variables have been added to our SCSS config.

Aichnerc committed:
Improve design

The general design of buttons and the profile page has been improved.

Aichnerc committed:
Fix Calendar3D

The code of the strike and contrib boxes has been greatly improved and does not longer suck. The design of those boxes has been improved. The max height of the chart bars has been reduced.

Aichnerc committed:
Improve Language chart

There is now a grey chart if there is no language found.

Aichnerc committed:
Improve design

The design of buttons and pinned items has been improved.

Aichnerc committed:
Add profile button to main navbar

The profile button has been moved.

Aichnerc committed:
Improve profile button on navbar

The padding has been adjusted.

Aichnerc committed:
Fix day and days seperation

The streaks are now marked as days or day.

Aichnerc committed:
Improve project cards

The slash for owners has been replaced with an @.

schettn committed:
Implement Movable Pool Save Functionallity

Now when pressing the save button the pool gets stored in the platformData.

schettn committed:
Replace Local Storage Save with Pool

Instead of local storage the indexArray is stored in a passed in object.

schettn committed:
Improve Read Cache Action

The loading bar is disabled now because it interferes with the caching loading bar.
Now the user object is set every time a key is missing.

schettn committed:
Improve Write Cache Action

Now the platform data is stringified before cached.

schettn committed:
Merge branch 'improve-basics-aa3' into add-movable-aa1

schettn committed:
Improve Movable Boundary

Now the item order is saved on swapping

schettn committed:
Improve Code Quality

The code quality has been improved due to codacy check.

pinterid committed:
Fix Codacy

The issues marked by Codacy were resolved.

pinterid committed:
Fix Codacy

Issues marked by Codacy were resolved.

schettn committed:
Improve code quality

The code quality has been improved due to a request of @pinterid.

Aichnerc committed:
Improve wording

None has been replaced with Unknown in the project item.

Aichnerc committed:
Improve stability

If there is no owner for a project, the new owner is unknown.

schettn committed:
Re-add loading bar styling

The loading bar styling, which was previously accidentally removed, is now available
again. This will show the loading bar again.

schettn committed:
Merge pull request #89 from snek-at/add-movable-aa1

Add movable aa1 to improve basics aa3

schettn committed:
Merge branch 'improve-basics-aa3' into improve-design-ca1

schettn committed:
Improve code quality

Some scss stuff has been fixed due to a codacy request.

schettn committed:
Apply suggestions from code review

Co-authored-by: Aichner Christian <contact@aichner-christian.com>

schettn committed:
Apply suggestions from code review

Aichnerc committed:
Merge pull request #91 from snek-at/improve-design-ca1

Improve design ca1 to improve basics aa3

schettn committed:
Improve code quality

The code quality has been improved due to request of @aichnerc @pinterid @kleberbaum.

Aichnerc commented on src/App.js:
```diff
@@ -21,287 +22,42 @@ import { Footer, Navbar } from "./components/molecules";
 //#region > Exports
-//> Default Class
+// Got access to the history object’s properties and the closest
+// <Route>'s match.
```
This is not the correct formatting for multiline comments.

Aichnerc commented on src/components/molecules/Navbar/index.jsx:
```diff
@@ -59,136 +64,144 @@ class Navbar extends React.Component {
 //#endregion
 
 //#region > PropTypes
+// is it required to write down all stuff in map...toProps??
```
What is this comment?

Aichnerc commented on src/store/actions/generalActions.js:
```diff
@@ -0,0 +1,161 @@
+      const intel = getIntel();
+
+      intel.appendList(sourceList);
+      dispatch({
```
I'd handle dispatches the same as returns and insert a new line above it.

schettn committed:
Upgrade TalkUploadModal with Redux

Due to a merging problem the talk upload modal did not use redux and therefore
did not work anymore.  Redux was added in order to get it back working.

Aichnerc reviewed (changes requested):
Found a few things.

schettn committed:
Improve code quality

The code quality has been improved due to a reuqest of @aichnerc.

Aichnerc reviewed (approved):
LGTM 🍕

schettn committed:
Add loggedUser Object on Read Cache

Now when reading the cache the loggedUser is set with fetchUser data if 
the read user is the same as the loggedUser.

kleberbaum commented on src/components/organisms/tabs/TalksTab/index.jsx:
```diff
@@ -195,17 +202,36 @@ class Talks extends React.Component {
+ * the store, and the functions it can use to dispatch actions to the store.
+ * Got access to the history object’s properties and the closest
+ * <Route>'s match.
+ */
```
```suggestion
/**
 * Provides its connected component with the pieces of the data it needs from
 * the store, and the functions it can use to dispatch actions to the store.
 *
 * Got access to the history object’s properties and the closest
 * <Route>'s match.
 */
```

kleberbaum commented on src/components/pages/ProfilePage/index.jsx:
```diff
@@ -84,9 +129,32 @@ class ProfilePage extends React.Component {
+ * the store, and the functions it can use to dispatch actions to the store.
+ * Got access to the history object’s properties and the closest
+ * <Route>'s match.
+ */
```
```suggestion
/**
 * Provides its connected component with the pieces of the data it needs from
 * the store, and the functions it can use to dispatch actions to the store.
 *
 * Got access to the history object’s properties and the closest
 * <Route>'s match.
 */
```

kleberbaum commented on src/components/pages/ProfilePage/profile.scss:
```diff
@@ -140,6 +162,7 @@
     }
   }
+
   // General things
```
?
```suggestion
```

kleberbaum commented on src/index.js:
```diff
@@ -21,15 +32,50 @@ import "./index.scss";
+const INTEL = new Intel();
+
+//#TODO
+// Must be moved to INTEL in future?
```
Mhh, good question should be part of a meeting.

kleberbaum commented on src/index.js:
```diff
@@ -21,15 +32,50 @@ import "./index.scss";
+  applyMiddleware(
+    loadingBarMiddleware(),
+    thunk.withExtraArgument({
+      getIntel, // Intel
```
Is this a valid comment?
```suggestion
      // Intel
      getIntel,
```

Aichnerc replied:
It makes sense to comment the those params the way proposed by @schettn , yes.

kleberbaum commented on src/store/reducers/index.js:
```diff
@@ -0,0 +1,28 @@
+//> Reducers
+import { loadingBarReducer } from "react-redux-loading-bar";
+
+//> Reducers
```
```//> Reducers``` is used multiple times.

kleberbaum commented on src/store/reducers/index.js:
```diff
@@ -0,0 +1,28 @@
+import authReducer from "./authReducers";
+import generalReducer from "./generalReducers";
+import userReducer from "./userReducers";
+
```
```suggestion
```

kleberbaum commented on src/store/reducers/index.js:
```diff
@@ -0,0 +1,28 @@
+  loadingBar: loadingBarReducer, // Loading bar
+  auth: authReducer, // User authentication
+  general: generalReducer, // General data
+  user: userReducer, // User data
```
I really don't like this comment style.

kleberbaum commented on src/components/atoms/SearchBar/index.jsx:
```diff
@@ -108,13 +108,28 @@ class SearchBar extends React.Component {
+ * the store, and the functions it can use to dispatch actions to the store.
+ * Got access to the history object’s properties and the closest
+ * <Route>'s match.
+ */
```
```suggestion
/**
 * Provides its connected component with the pieces of the data it needs from
 * the store, and the functions it can use to dispatch actions to the store.
 *
 * Got access to the history object’s properties and the closest
 * <Route>'s match.
 */
```

kleberbaum commented on src/components/molecules/Navbar/index.jsx:
```diff
@@ -59,136 +64,146 @@ class Navbar extends React.Component {
+ * the store, and the functions it can use to dispatch actions to the store.
+ * Got access to the history object’s properties and the closest
+ * <Route>'s match.
+ */
```
```suggestion
/**
 * Provides its connected component with the pieces of the data it needs from
 * the store, and the functions it can use to dispatch actions to the store.
 *
 * Got access to the history object’s properties and the closest
 * <Route>'s match.
 */
```

kleberbaum reviewed (changes requested):
Should be ok to merge after a few changes.

schettn committed:
Improve code quality

The code quality has been improved due to a request of @kleberbaum.

schettn reviewed (approved):
Yep

kleberbaum reviewed (approved):
Should be ok to merge.
